x11rb/src/rust_connection/stream.rs

```diff
@@ -550,29 +547,29 @@ impl Stream for DefaultStream {
 }
 
 #[cfg(any(target_os = "linux", target_os = "android"))]
-fn connect_abstract_unix_stream(path: &[u8]) -> nix::Result<RawFdContainer> {
-    use nix::fcntl::{fcntl, FcntlArg, OFlag};
-    use nix::sys::socket::{connect, socket, AddressFamily, SockFlag, SockType, UnixAddr};
-
-    let socket = socket(
-        AddressFamily::Unix,
-        SockType::Stream,
-        SockFlag::SOCK_CLOEXEC,
-        None,
+fn connect_abstract_unix_stream(
+    path: &[u8],
+) -> std::result::Result<RawFdContainer, rustix::io::Errno> {
+    use rustix::fs::{fcntl_getfl, fcntl_setfl, OFlags};
+    use rustix::net::{
+        connect_unix, socket_with, AddressFamily, Protocol, SocketAddrUnix, SocketFlags, SocketType,
+    };
+
+    let socket = socket_with(
+        AddressFamily::UNIX,
+        SocketType::STREAM,
+        SocketFlags::CLOEXEC,
+        Protocol::default(),
     )?;
 
     // Wrap it in a RawFdContainer. Its Drop impl makes sure to close the socket if something
     // errors out below.
```

Comment:
Well... this comment could be updated. The `Drop` impl of `RawFdContainer` is no longer relevant for this, since this is already an `OwnedFd`, right? (In fact, there is no longer a `Drop` impl)